analyzers/src/SonarAnalyzer.Common/Helpers/CountComparisonSyntax.cs

```diff
@@ -0,0 +1,94 @@
+﻿using System;
+using System.Linq;
+
+namespace SonarAnalyzer.Helpers
+{
+    public readonly struct CountComparisonSyntax : IEquatable<CountComparisonSyntax>
+    {
+        public enum Kind
+        {
+            Equals,
+            NotEquals,
+            LessThan,
+            LessThanOrEqual,
+            GreaterThan,
+            GreaterThanOrEqual,
+        }
+
+        public CountComparisonSyntax(int? left, Kind @operator, int? right)
+        {
+            Left = left;
+            Operator = @operator;
+            Right = right;
+        }
+        public int? Left { get; }
+        public int? Right { get; }
+        public Kind Operator { get; }
+
+        public bool ExpressionIsLeft() => !Left.HasValue;
+        public bool IsValid() => Left.HasValue ^ Right.HasValue;
+        public bool IsEmpty() => Empties.Contains(Normalized());
+        public bool IsAny() => Anys.Contains(Normalized());
+        public bool IsEmptyOrAny() => IsEmpty() || IsAny();
+        public bool IsWithoutMeaning()
+        {
+            if (IsValid())
+            {
+                var normalized = Normalized();
+                var val = normalized.Right.Value;
+                if (val < 0)
+                {
+                    return true;
+                }
+                else if (val == 0)
+                {
+                    return normalized.Operator == Kind.LessThan
+                        || normalized.Operator == Kind.GreaterThanOrEqual;
+                }
+                else
+                {
+                    return false;
+                }
+            }
+            return false;
+        }
+
+        public CountComparisonSyntax Normalized() =>
+           ExpressionIsLeft()
+           ? this
+           : new CountComparisonSyntax(left: Right, Mirror(Operator), right: Left);
+
+        public override bool Equals(object obj) => obj is CountComparisonSyntax other && Equals(other);
+        public bool Equals(CountComparisonSyntax other) =>
+            Left == other.Left
+            && Right == other.Right
+            && Operator == other.Operator;
+        public override int GetHashCode() =>
+            Left.GetHashCode()
+            ^ (Right.GetHashCode() << 14)
+            ^ (Operator.GetHashCode() << 28);
+        public override string ToString() => $"{Left} {Operator} {Right}";
+
+        private static readonly CountComparisonSyntax[] Empties = new[]
+        {
+            new CountComparisonSyntax(default, Kind.Equals, 0),
```

Comment:
Please use `null` instead of `default` for all nullable.
```suggestion
            new CountComparisonSyntax(null, Kind.Equals, 0),
```